examples/step-98/step-98.cc

```diff
@@ -0,0 +1,1106 @@
+/* ------------------------------------------------------------------------
+ *
+ *
+ * Authors: Michal Wichrowski, Heidelberg University,
+ * 2025
+ */
+
+
+// First include the necessary files from the deal.II library.
+
+#include <deal.II/base/function.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/timer.h>
+
+#include <deal.II/dofs/dof_renumbering.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_tools.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/lac/affine_constraints.h>
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/la_parallel_vector.h>
+#include <deal.II/lac/precondition.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/solver_gmres.h>
+#include <deal.II/lac/solver_richardson.h>
+#include <deal.II/lac/tensor_product_matrix.h>
+
+#include <deal.II/matrix_free/fe_evaluation.h>
+#include <deal.II/matrix_free/matrix_free.h>
+#include <deal.II/matrix_free/operators.h>
+#include <deal.II/matrix_free/fe_patch_evaluation.h>
+
+
+
+#include <deal.II/multigrid/mg_coarse.h>
+#include <deal.II/multigrid/mg_matrix.h>
+#include <deal.II/multigrid/mg_smoother.h>
+#include <deal.II/multigrid/mg_tools.h>
+#include <deal.II/multigrid/mg_transfer_matrix_free.h>
+#include <deal.II/multigrid/multigrid.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/vector_tools.h>
+
+// This includes the utilities for the efficient implementation of
+// matrix-free patch-smoothing methods.
+#include <deal.II/matrix_free/patch_storage.h>
+#include <deal.II/matrix_free/patch_distributors.h>
+#include <deal.II/matrix_free/patch_smoother_base.h>
+
+
+
+#include <fstream>
+#include <iostream>
+
+
+
+namespace Operators
+{
+  using namespace dealii;
+
+  /**
+   *
+   * This class implements the Laplace operator $-\Delta u$ as in step-37 using
+   * matrix-free techniques based on the deal.II::MatrixFree framework. It
```

Comment:
```suggestion
   * matrix-free techniques based on deal.II's MatrixFree framework. It
```
(I think otherwise doxygen will not link with the class, and in any case it might be confusing because `deal.II` is not the namespace name, even though `::` might suggest so.)